System.NullReferenceException: Object reference not set to an instance of an object in mdoc/Mono.Documentation/Updater/DocumentationEnumerator.cs(235,53)

While building mdoc.exe from https://github.com/mono/mono (mono-6.6.0.161) on Fedora Linux with this patch https://gist.github.com/NattyNarwhal/b4945bc5e1f5213d5eaa8303261416bd#file-doctools-diff I'm encountering the following error:

```
[ERROR] FATAL UNHANDLED EXCEPTION: Mono.CSharp.InternalErrorException: ../../../external/api-doc-tools/mdoc/Mono.Documentation/Updater/DocumentationEnumerator.cs(240,53): Mono.Documentation.Updater.DocumentationEnumerator.GetReflectionMembersCore(Mono.Cecil.TypeDefinition, string, string) ---> Mono.CSharp.InternalErrorException: (1,1): ---> Mono.CSharp.InternalErrorException: ../../../external/api-doc-tools/mdoc/Mono.Documentation/Updater/DocumentationEnumerator.cs(241,9): Mono.Documentation.Updater.DocumentationEnumerator.<GetReflectionMembersCore>c__Iterator2.<iface>__10 ---> System.NullReferenceException: Object reference not set to an instance of an object
  at IKVM.Reflection.Signature.WriteType (IKVM.Reflection.Emit.ModuleBuilder module, IKVM.Reflection.Writer.ByteBuffer bb, IKVM.Reflection.Type type) [0x000b6] in <478ef8dac9cd4b8f8880543e113182e2>:0 
  at IKVM.Reflection.FieldSignature.WriteSig (IKVM.Reflection.Emit.ModuleBuilder module, IKVM.Reflection.Writer.ByteBuffer bb) [0x00014] in <478ef8dac9cd4b8f8880543e113182e2>:0 
  at IKVM.Reflection.Emit.FieldBuilder..ctor (IKVM.Reflection.Emit.TypeBuilder type, System.String name, IKVM.Reflection.Type fieldType, IKVM.Reflection.CustomModifiers customModifiers, IKVM.Reflection.FieldAttributes attribs) [0x00062] in <478ef8dac9cd4b8f8880543e113182e2>:0 
  at IKVM.Reflection.Emit.TypeBuilder.__DefineField (System.String fieldName, IKVM.Reflection.Type type, IKVM.Reflection.CustomModifiers customModifiers, IKVM.Reflection.FieldAttributes attributes) [0x00000] in <478ef8dac9cd4b8f8880543e113182e2>:0 
  at IKVM.Reflection.Emit.TypeBuilder.DefineField (System.String fieldName, IKVM.Reflection.Type type, IKVM.Reflection.Type[] requiredCustomModifiers, IKVM.Reflection.Type[] optionalCustomModifiers, IKVM.Reflection.FieldAttributes attributes) [0x0000b] in <478ef8dac9cd4b8f8880543e113182e2>:0 
  at Mono.CSharp.Field.Define () [0x00076] in <478ef8dac9cd4b8f8880543e113182e2>:0 
  at Mono.CSharp.TypeDefinition.DoDefineMembers () [0x0033a] in <478ef8dac9cd4b8f8880543e113182e2>:0 
   --- End of inner exception stack trace ---
  at Mono.CSharp.TypeDefinition.DoDefineMembers () [0x0035b] in <478ef8dac9cd4b8f8880543e113182e2>:0 
  at Mono.CSharp.ClassOrStruct.DoDefineMembers () [0x000e9] in <478ef8dac9cd4b8f8880543e113182e2>:0 
  at Mono.CSharp.CompilerGeneratedContainer.DoDefineMembers () [0x00033] in <478ef8dac9cd4b8f8880543e113182e2>:0 
  at Mono.CSharp.StateMachine.DoDefineMembers () [0x0002c] in <478ef8dac9cd4b8f8880543e113182e2>:0 
  at Mono.CSharp.IteratorStorey.DoDefineMembers () [0x0022a] in <478ef8dac9cd4b8f8880543e113182e2>:0 
  at Mono.CSharp.TypeDefinition.Define () [0x00012] in <478ef8dac9cd4b8f8880543e113182e2>:0 
  at Mono.CSharp.ExplicitBlock.DefineStoreyContainer (Mono.CSharp.EmitContext ec, Mono.CSharp.AnonymousMethodStorey storey) [0x004db] in <478ef8dac9cd4b8f8880543e113182e2>:0 
  at Mono.CSharp.ParametersBlock.Emit (Mono.CSharp.EmitContext ec) [0x0001c] in <478ef8dac9cd4b8f8880543e113182e2>:0 
  at Mono.CSharp.ToplevelBlock.Emit (Mono.CSharp.EmitContext ec) [0x00026] in <478ef8dac9cd4b8f8880543e113182e2>:0 
   --- End of inner exception stack trace ---
  at Mono.CSharp.ToplevelBlock.Emit (Mono.CSharp.EmitContext ec) [0x000df] in <478ef8dac9cd4b8f8880543e113182e2>:0 
  at Mono.CSharp.MethodData.Emit (Mono.CSharp.TypeDefinition parent) [0x00092] in <478ef8dac9cd4b8f8880543e113182e2>:0 
  at Mono.CSharp.MethodOrOperator.Emit () [0x0021c] in <478ef8dac9cd4b8f8880543e113182e2>:0 
  at Mono.CSharp.Method.Emit () [0x0012c] in <478ef8dac9cd4b8f8880543e113182e2>:0 
   --- End of inner exception stack trace ---
  at Mono.CSharp.Method.Emit () [0x00141] in <478ef8dac9cd4b8f8880543e113182e2>:0 
  at Mono.CSharp.TypeDefinition.Emit () [0x002f5] in <478ef8dac9cd4b8f8880543e113182e2>:0 
  at Mono.CSharp.ClassOrStruct.Emit () [0x00025] in <478ef8dac9cd4b8f8880543e113182e2>:0 
  at Mono.CSharp.Class.Emit () [0x00000] in <478ef8dac9cd4b8f8880543e113182e2>:0 
  at Mono.CSharp.TypeDefinition.EmitContainer () [0x0000f] in <478ef8dac9cd4b8f8880543e113182e2>:0 
  at Mono.CSharp.TypeContainer.EmitContainer () [0x0001e] in <478ef8dac9cd4b8f8880543e113182e2>:0 
  at Mono.CSharp.NamespaceContainer.EmitContainer () [0x00007] in <478ef8dac9cd4b8f8880543e113182e2>:0 
  at Mono.CSharp.TypeContainer.EmitContainer () [0x0001e] in <478ef8dac9cd4b8f8880543e113182e2>:0 
  at Mono.CSharp.NamespaceContainer.EmitContainer () [0x00007] in <478ef8dac9cd4b8f8880543e113182e2>:0 
  at Mono.CSharp.TypeContainer.EmitContainer () [0x0001e] in <478ef8dac9cd4b8f8880543e113182e2>:0 
  at Mono.CSharp.ModuleContainer.EmitContainer () [0x0009c] in <478ef8dac9cd4b8f8880543e113182e2>:0 
  at Mono.CSharp.AssemblyDefinition.Emit () [0x00098] in <478ef8dac9cd4b8f8880543e113182e2>:0 
  at Mono.CSharp.AssemblyDefinitionStatic.Emit () [0x00126] in <478ef8dac9cd4b8f8880543e113182e2>:0 
  at Mono.CSharp.Driver.Compile () [0x002f3] in <478ef8dac9cd4b8f8880543e113182e2>:0 
  at Mono.CSharp.Driver.Main (System.String[] args) [0x00052] in <478ef8dac9cd4b8f8880543e113182e2>:0 
make[9]: *** [../../build/executable.make:142: ../../class/lib/net_4_x-linux/mdoc.exe] Error 1
```
I didn't manage to fix it, may you help?

--- a/mdoc/Mono.Documentation/Updater/DocumentationEnumerator.cs
+++ b/mdoc/Mono.Documentation/Updater/DocumentationEnumerator.cs
@@ -341,7 +341,7 @@ namespace Mono.Documentation.Updater
                             var method = ((PropertyDefinition) mr).GetMethod ?? ((PropertyDefinition) mr).SetMethod;
                             if (method?.Overrides != null && method.Overrides.Any())
                             {
-                                DocUtils.GetInfoForExplicitlyImplementedMethod(method, out var iface, out var ifaceMethod);
+                                DocUtils.GetInfoForExplicitlyImplementedMethod(method, out TypeReference iface, out MethodReference ifaceMethod);
                                 var newName = DocUtils.GetMemberForProperty(ifaceMethod.Name);
                                 if (newName == memberName && verifyInterface(mr) && docName.Contains (iface.Name))
                                     yield return mr;

